src/devices/laserFromPCL/laserFromPCL.cpp

```diff
@@ -0,0 +1,480 @@
+#endif
+
+    m_min_distance = 0.1; //m
+    m_max_distance = 5.0;  //m
+    m_floor_height = 0.1; //m
+    m_ceiling_height = 2; //m
+    m_publish_ros_pc =true;
+    m_ground_frame_id = "/ground_frame";
+    m_camera_frame_id = "/depth_camera_frame";
+
+    bool bpc = config.check("Z_CLIPPING_PLANES");
+    if (bpc != false)
+    {
+        yarp::os::Searchable& pointcloud_clip_config = config.findGroup("Z_CLIPPING_PLANES");
+        if (pointcloud_clip_config.check("floor_height"))   {m_floor_height = pointcloud_clip_config.find("floor_height").asFloat64();}
+        if (pointcloud_clip_config.check("ceiling_height")) {m_ceiling_height=pointcloud_clip_config.find("ceiling_height").asFloat64();}
+        if (pointcloud_clip_config.check("ground_frame_id")) {m_ground_frame_id = pointcloud_clip_config.find("ground_frame_id").asString();}
+        if (pointcloud_clip_config.check("camera_frame_id")) {m_camera_frame_id = pointcloud_clip_config.find("camera_frame_id").asString();}
+
+    }
+    yInfo() <<"Z clipping planes (floor,ceiling) have been set to ("<< m_floor_height <<","<< m_ceiling_height <<")";
+
+    if (this->parse(config) == false)
+    {
+        yError() << "LaserFromExternalPort: error parsing parameters";
+        return false;
+    }
+
+    //open the tc client
+    Property tcprop;
+    if(!config.check("TRANSFORM_CLIENT"))
+    {
+        yError() << "missing TRANSFORM_CLIENT section in configuration file!";
+        return false;
+    }
+    tcprop.fromString(config.findGroup("TRANSFORM_CLIENT").toString());
+    tcprop.put("device", "transformClient");
+    //prop.put("local", "/"+m_module_name + "/TfClient");
+    //prop.put("remote", "/transformServer");
+
+    m_tc_driver.open(tcprop);
+    if (!m_tc_driver.isValid())
+    {
+        yError("Error opening PolyDriver check parameters");
+        return false;
+    }
+    m_tc_driver.view(m_iTc);
+    if (!m_iTc)
+    {
+        yError("Error opening iFrameTransform interface. Device not available");
+        return false;
+    }
+    yarp::os::Time::delay(0.1);
+
+
+    //open the rgbd client
+    Property prop;
+    if(!config.check("RGBD_SENSOR_CLIENT"))
+    {
+        yError() << "missing RGBD_SENSOR_CLIENT section in configuration file!";
+        return false;
+    }
+    prop.fromString(config.findGroup("RGBD_SENSOR_CLIENT").toString());
+    prop.put("device", "RGBDSensorClient");
+    m_rgbd_driver.open(prop);
+    if (!m_rgbd_driver.isValid())
+    {
+        yError("Error opening PolyDriver check parameters");
+        return false;
+    }
+    m_rgbd_driver.view(m_iRGBD);
+    if (!m_iRGBD)
+    {
+        yError("Error opening iRGBD interface. Device not available");
+        return false;
+    }
+    yarp::os::Time::delay(0.1);
+
+    //get parameters data from the camera
+    m_depth_width = m_iRGBD->getDepthWidth();
+    m_depth_height = m_iRGBD->getDepthHeight();
+    bool propintr  = m_iRGBD->getDepthIntrinsicParam(m_propIntrinsics);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'propintr' is assigned a value that is never used.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=5040974)